saleor/graphql/core/types/taxes.py

```diff
@@ -389,13 +402,47 @@ def calculate_checkout_discounts(checkout_info):
             )
 
         def map_discounts(discounts):
```

Comment:
Can we simplify the logic from this function? I mean, right now you have one `for` for handling manuals, then second for for handling the promotion-discount and voucher. We can simplify this and have single loop. This would be faster, and simpler to read. Something like this:
```python
        def map_discounts(discounts):
            taxable_discounts = []
            for discount in discounts:
                subtotal_discount, shipping_discount = 0, 0
                if discount.type == DiscountType.MANUAL:
                    subtotal = root.subtotal.net
                    shipping = root.base_shipping_price
                    subtotal_discount, shipping_discount = split_manual_discount(
                        discount, subtotal, shipping
                    )
                if is_order_level_voucher(discount.voucher) or discount.type == DiscountType.ORDER_PROMOTION:
                    subtotal_discount = discount.amount

                if subtotal_discount:
                    taxable_discounts.append({
                        "name": discount.name,
                        "amount": discount.amount,
                        "type": TaxableObjectDiscountTypeEnum.SUBTOTAL,
                    })
                if shipping_discount:
                    taxable_discounts.append({
                        "name": discount.name,
                        "amount": discount.amount,
                        "type": TaxableObjectDiscountTypeEnum.SHIPPING,
                    })

            return taxable_discounts
```